src/App.tsx

```diff
@@ -1,24 +1,157 @@
+          recipe.thumbnail.show = false;
+        }
+      });
+      setAllRecipes(data);
+    }
+  };
+
+  useEffect(() => {
+    getAllRecipes();
+  }, []);
+
+  const onChange = (
+    e: string | string[],
+    index: number,
+    image: string,
+    show: boolean
+  ) => {
+    const tempPage: Recipe[] = [...allRecipes];
+    const recipe = { ...tempPage[index] };
+    if (image !== "") {
+      if (show === true) {
+        recipe.thumbnail.show = false;
+      } else {
+        recipe.thumbnail.show = true;
+      }
+      tempPage[index] = recipe;
+      setAllRecipes(tempPage);
+    } else {
+      recipe.thumbnail.show = false;
+      tempPage[index] = recipe;
+      setAllRecipes(tempPage);
+    }
+  };
+
+  const { Panel } = Collapse;
+
+  const genExtra = () => (
+    <EditOutlined
+      onClick={(event) => {
+        // If you don't want click extra trigger collapse, you can prevent this:
+        event.stopPropagation();
+      }}
+    />
+  );
+
+  const getRecommendationPic = (id: string) => {
+    var recommendation = allRecipes.find((item) => item.id === parseInt(id));
+    if (recommendation !== undefined) {
+      return recommendation.thumbnail.image;
+    }
+  };
+
+  const getRecommendationName = (id: string) => {
+    var recommendation = allRecipes.find((item) => item.id === parseInt(id));
+    if (recommendation !== undefined) {
+      return recommendation.name;
+    }
+  };
+
+  const displayIngredient = (ingredient: string, starred: boolean) => {
+    if (starred === true) {
+      return ingredient;
+    }
+  };
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <h1>Recipe App</h1>
+
+      <div className="Ipad">
+        {allRecipes.map((item, index) => (
+          <div className="Recipes" key={index}>
+            <>
+              {item.thumbnail.show && (
+                <Image width={150} src={item.thumbnail.image} preview={false} />
+              )}
+            </>
+            <div className="RecipeBox">
+              <Collapse
+                onChange={(e) =>
+                  onChange(e, index, item.thumbnail.image, item.thumbnail.show)
+                }
+                expandIconPosition="end"
+              >
+                <Panel header={item.name} key={index} extra={genExtra()}>
+                  <div className="DescriptionBox">{item.description}</div>
+                  <div className="FirstHalf">
+                    <div className="IngredientBox">
+                      <h2>Ingredients</h2>
+                      {item.ingredients.map((ingred, ingredIndex) => (
```

Comment:
Please change this to `ingredient`.  You generally don't want to abbreviate variable names except in certain rare circumstances